packages/snaps-rpc-methods/src/permitted/scheduleBackgroundEvent.ts

```diff
@@ -107,14 +131,23 @@ async function getScheduleBackgroundEventImplementation(
   try {
     const validatedParams = getValidatedParams(params);
 
-    const { date, request } = validatedParams;
+    const { request } = validatedParams;
+
+    let truncatedDate;
```

Comment:
It's cleaner to move this to a function and return the date, to avoid mutating variables.